dispatcher/pathbased.go

```diff
@@ -127,6 +128,17 @@ func (d *pathBasedDispatcher) buildOptionsHandler(cfg *config.Application, inner
 		}
 
 		rw.Header().Set("Allow", allow)
+
+		body, err := ioutil.ReadAll(recorder.Body)
+		if err != nil {
+			d.log.Errorf("error while reading response body: %s", err)
+			rw.WriteHeader(500)
```

Comment:
Maybe use 502 or 503 as status code?

Reply:
In other parts of the program, for the same error 500 is used, so for consistency 500 should be used.